query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/PrometheusUtils.java

```diff
@@ -0,0 +1,14 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import org.hypertrace.core.query.service.QueryRequestUtil;
+import org.hypertrace.core.query.service.api.Expression;
+
+class PrometheusUtils {
+  static String getColumnNameForMetricFunction(Expression functionExpression) {
+    String functionName = functionExpression.getFunction().getFunctionName().toUpperCase();
+    String columnName =
+        QueryRequestUtil.getLogicalColumnNameForSimpleColumnExpression(
+            functionExpression.getFunction().getArguments(0));
+    return String.join(":", functionName, columnName);
```

Comment:
This makes me nervous - a lot of the work on the attribute expressions feature has been in fixing similar code where we generate aliases or use attribute IDs as column names in the result metadata and ignore the requested alias. Since this is ignoring the requested alias, is it for internal use only or does this show up in the response?

Reply:
It's for internal mapping and internal use only (It doesn't show up in response). As we have to fire multiple queries for the same QS query, and if that query has two metrics function on the same attributed, we needed a unique mapping.

e.g 
```
select SERVICE.service_id, sum(SERVICE.numCalls), avg(SERVICE.numCalls).
```

As in the above example, the same attribute id is used for two different metric functions. It wasn't the issue in the case of PinotBase response where it can be mapped using column index. But, here, we will have two PromQL responses - one for `sum(SERVICE.numCalls)` and one for `avg(SERVICE.numCalls)`. So, internally, we need a unique map of the response, so we are using `<funcname>:<attributeId>` internally for response building.

Reply:
Got it; In other places, we were building the response metadata like this, rather than using the alias. As long as it's switching back to that before returning, we're good.